### PR TITLE
[idrac_system_info] - Enhancement to support iDRAC10

### PR DESCRIPTION
**Pull request by Saksham-Nautiyal**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: plugins/modules/idrac_firmware_info.py [hunk 1]
```diff
@@ -107,32 +107,53 @@
 
 import json
 from ansible_collections.dellemc.openmanage.plugins.module_utils.dellemc_idrac import iDRACConnection, idrac_auth_params
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/idrac_firmware_info.py:109]: remove this import if you are using iDRACRedfishAPI

File: plugins/modules/idrac_firmware_info.py [hunk 2]
```diff
@@ -107,32 +107,53 @@
...
+from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI
+from ansible_collections.dellemc.openmanage.plugins.module_utils.utils import remove_key
 from ansible.module_utils.basic import AnsibleModule
 from ansible.module_utils.urls import ConnectionError, SSLValidationError
-from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from urllib.error import URLError, HTTPError
 
+GET_IDRAC_FIRMWARE_DETAILS_URI= "/redfish/v1/UpdateService/Oem/Dell/DellSoftwareInventory?$select=Members"
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/idrac_firmware_info.py:116]: is it for iDRAC 10, if yes explicitly mention in URI name, so it's clear.

File: plugins/modules/idrac_firmware_info.py [hunk 3]
```diff
@@ -107,32 +107,53 @@
...
+GET_IDRAC_FIRMWARE_URI= "/redfish/v1/UpdateService/Oem/Dell/DellSoftwareInventory"
+
+def get_idrac_firmware_info(idrac, module):
+    try:
+        response = idrac.invoke_request(method='GET',uri=GET_IDRAC_FIRMWARE_URI)
+
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/idrac_firmware_info.py:122]: same what if response status code is not 200?

File: plugins/modules/idrac_firmware_info.py [hunk 4]
```diff
@@ -107,32 +107,53 @@
...
-                     firmware_info=msg)
+    module.exit_json(
+        msg="Successfully fetched the firmware inventory details.",
+        firmware_info = firmware_info
+    )
 
 
 if __name__ == '__main__':
-    main()
+    main()
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/idrac_firmware_info.py:159]: Add new line

File: plugins/modules/idrac_system_info.py [hunk 5]
```diff
@@ -3,7 +3,7 @@
 
 #
 # Dell OpenManage Ansible Modules
-# Version 7.1.0
+# Version 9.8.0
 # Copyright (C) 2021-2022 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/idrac_system_info.py:7]: There's no update in UT?

**Comment by Saksham-Nautiyal** [comment]
closing due to divergent branches

**Pull request closed without merging**